index.js

```diff
@@ -36,33 +37,60 @@ var gulpSass = function gulpSass(options) {
       opts.omitSourceMapUrl = true;
     }
 
-    callback = function(error, obj) {
-      if (error) {
+    if (sync !== true) {
+      callback = function(error, obj) {
+        if (error) {
+          return cb(new gutil.PluginError(
+            PLUGIN_NAME, error.message + ' ' + gutil.colors.cyan('line ' + error.line) + ' in ' + gutil.colors.magenta(error.file)
+          ));
+        }
+        // Build Source Maps!
+        if (obj.map) {
+          applySourceMap(file, JSON.parse(obj.map.toString()));
+        }
+
+        file.contents = obj.css;
+        file.path = gutil.replaceExtension(file.path, '.css');
+
+        cb(null, file);
+      };
+
+      sass.render(opts, callback);
+    }
+    else {
+      try {
+        result = sass.renderSync(opts);
+
+        // Build Source Maps!
+        if (result.map) {
+          applySourceMap(file, JSON.parse(result.map.toString()));
+        }
+
+        file.contents = result.css;
+        file.path = gutil.replaceExtension(file.path, '.css');
+
+        cb(null, file);
```

Comment:
Although considering how little code it is and the complexity of the DRY abstraction, is it really worth it?